src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -509,14 +625,20 @@ def _get_app_env_and_group(
     if not resource_group.name and not resource_group.exists:
         matched_apps = [c for c in list_containerapp(cmd) if c["name"].lower() == name.lower()]
         if env.name:
-            matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["managedEnvironmentId"])["name"].lower() == env.name.lower()]
+            matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["name"].lower() == env.name.lower()]
+        if env.resource_type:
+            matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["resource_type"].lower() == env.resource_type.lower()]
+        if env.custom_location.name:
+            matched_apps = [c for c in matched_apps if c["extendedLocation"]["name"].lower() == env.custom_location.name.lower()]
         if location:
-            matched_apps = [c for c in matched_apps if c["location"].lower() == location.lower()]
+            matched_apps = [c for c in matched_apps if format_location(c["location"]) == format_location(location)]
```

Comment:
with so many if here, finally the matched app will be filtered by location info even there is environment info matched.
Consider the priority of these if conditions.
If the environment is provided by cx, and the app's env is not match, what will you do?

Reply:
I didn't change the logic of filtering by location.

format_location is for fix bug: eastus' and 'East us' is same region.

Reply:
"If the environment is provided by cx, and the app's env is not match, what will you do?"
In fact, it could be found:
def get_container_app_if_exists(cmd, resource_group_name, name):
    app = None
    try:
        app = ContainerAppClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
    except:  # pylint: disable=bare-except
        pass
    return app